### PR TITLE
Nginx config by terraform

### PR DESCRIPTION
**Pull request by tkostyuk-nf**
Manage Nginx configs by terraform to allow unique configuration for production and staging.

### CODE REVIEW

File: infra/packer/nginx/sites-available/staging.network.holmusk.com [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    proxy_pass http://127.0.0.1:5000;
+    proxy_http_version 1.1;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection "upgrade";
+    proxy_set_header Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_set_header X-Forwarded-Host $host;
+    proxy_set_header X-Forwarded-Port $server_port;
```
Comment (turboMaCk) [inline infra/packer/nginx/sites-available/staging.network.holmusk.com:18]: Q: don't you know how far is the support of spec compliant [Forwarded](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) header? I know X-Forwarded-* is become sort of convention but perhaps it would be nice to slowly switch to spec compliant header.

File: infra/packer/nginx/sites-available/staging.network.holmusk.com [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    proxy_cache_bypass $http_upgrade;
+    rewrite ^/backend(.*)$ $1 break;
```
Comment (turboMaCk) [inline infra/packer/nginx/sites-available/staging.network.holmusk.com:20]: I would consider extracting this block to file and importing it to both staging and production config, otherwise we are at risk of changing staging config forgetting about production one.

File: infra/packer/holmusk-network.json [hunk 3]
```diff
@@ -61,7 +61,8 @@
         "sudo systemctl enable holmusk-network-database.service",
         "sudo systemctl enable holmusk-network-backend.service",
         "sudo docker pull neo4j:latest",
```
Comment (turboMaCk) [inline infra/packer/holmusk-network.json:63]: I know it's not in scope of this PR but wildcatting to latest doesn't seem like a good idea to me.

Comment (tkostyuk-nf) [reply infra/packer/holmusk-network.json:63]: Running neo4j database in a container was the fastest way to get a working solution and got something to present.
https://app.asana.com/0/0/1183461973551054/f task is about switching to external DB instance.
The external DB instance version will also define what neo4j docker image version to be used for CI.

File: infra/packer/nginx/sites-available/staging.network.holmusk.com (hunk 1 above)
Comment (arbus) [reply infra/packer/nginx/sites-available/staging.network.holmusk.com]: This is a blurb that I got from some source on how to setup nginx to proxy. At this point, its almost 4 years old, just being copied from 1 project to another.

I've never changed this since its always worked. Do you think we should start changing it? If so, we can create a seperate task to track this since all our projects with nginx use this particular config.

**Comment by tkostyuk-nf** [comment]
Deployed to staging so is working and verified.

File: infra/packer/nginx/sites-available/staging.network.holmusk.com (hunk 1 above)
Comment (turboMaCk) [reply infra/packer/nginx/sites-available/staging.network.holmusk.com]: We probably don't need this at all. This is used for letting application behind the proxy know about over which port, host and protocol the original request is made. It's usually usefull for stuff like redirecting from http to https in application logic (middleware) in cases one doesn't want to do such things via nginx. For long time `X-Forwarded-*` was used as there was not spec complaint way to do this. This is for instance what heroku uses. I think in our case we likely do such things directly in nginx similarly to how we did the cors setup for aia. I worked with these things a lot in past when I worked on switch from http to https compatible with heroku, aws and gcp. This is [nodejs example of how this can be used](github.com/turboMaCk/koa-sslify) and [here is Haskell implementation of same thing](https://github.com/turboMaCk/wai-enforce-https) so you have an idea for what this is usually used.

So tl;dr is that it's not a big deal. If this was just copied we probably don't even use it. I'm not sure how widely accepted `Forwarded` header is at this point. `X-Forwarded-*` headers were there for a while and people don't like changing stuff like this. Anyway the obvious advantage of Forwarded is that it's documented by specification while `X-Forwarded` header are just common convention which is not really standardized.
